utils/yoke/derive/src/lib.rs

```diff
@@ -123,33 +123,66 @@ fn yokeable_derive_impl(input: &DeriveInput) -> TokenStream2 {
                         // to `FieldTy: Yokeable` that need to be satisfied. We get them to be satisfied by requiring
                         // `FieldTy<'static>: Yokeable<FieldTy<'a>>`
                         if has_lt {
-                            let a_ty = replace_lifetime(&f.ty, custom_lt("'a"));
-                            yoke_bounds
-                                .push(parse_quote!(#fty: yoke::Yokeable<'a, Output = #a_ty>));
+                            let fty_a = replace_lifetime(&f.ty, custom_lt("'a"));
+                            yoke_bounds.push(
+                                parse_quote!(#fty_static: yoke::Yokeable<'a, Output = #fty_a>),
+                            );
                         } else {
-                            yoke_bounds.push(parse_quote!(#fty: yoke::Yokeable<'a, Output = #fty>));
+                            yoke_bounds.push(
+                                parse_quote!(#fty_static: yoke::Yokeable<'a, Output = #fty_static>),
+                            );
                         }
                     }
                     if has_ty || has_lt {
                         // By calling transform_owned on all fields, we manually prove
                         // that the lifetimes are covariant, since this requirement
                         // must already be true for the type that implements transform_owned().
                         quote! {
-                            <#fty as yoke::Yokeable<'a>>::transform_owned(#field)
+                            <#fty_static as yoke::Yokeable<'a>>::transform_owned(#field)
                         }
                     } else {
                         // No nested lifetimes, so nothing to be done
                         quote! { #field }
                     }
                 })
             });
+            let borrowed_body = structure.each(|binding| {
+                let f = binding.ast();
```

Comment:
Question: Why do you do `structure.each_variant(|vi| vi.construct(...))` above but `structure.each(|binding| binding.ast() )` down here?

Reply:
Above we're actually reconstructing the type, whereas here we're doing an assertion per field.

Think of it as the difference between a Clone derive (call clone on each field, put the type back together) vs a Debug derive (call Debug on each field, period)